src/plugins/inputs/msg.js

```diff
@@ -1,9 +1,71 @@
 "use strict";
 
-exports.commands = ["msg", "say"];
+const Chan = require("../../models/chan");
+const Msg = require("../../models/msg");
+
+exports.commands = ["query", "msg", "say"];
+
+function parseInput(cmd, args, chan) {
+	switch (cmd) {
+	case "msg":
+	case "query":
+		return args.shift();
+	default:
+		return chan.name;
+	}
+}
 
 exports.input = function(network, chan, cmd, args) {
-	const target = cmd === "msg" ? args.shift() : chan.name;
+	const target = parseInput(cmd, args, chan);
+
+	if (cmd === "query") {
+		if (typeof target === "undefined" || target.length === 0) {
```

Comment:
```suggestion
		if (!target) {
```

This would cover both of these cases, wouldn't it? It's done this way below for the `/msg`

Reply:
Correct but would have the issue with isQueryOpened that we discussed previously ?

Reply:
What issue?

Reply:
misunderstood, convention you are correct and it way cleaner. good catch